### PR TITLE
Aoc2024 day04

### PR DESCRIPTION
**Pull request by jbanass**

### CODE REVIEW

File: 2024/04/script.rb [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      end
+
+      accumulator
+    end
+  end
+
+  private
+
+  def check_word_from_coords(coords)
+    coords = coords.filter { |x, y| x >= 0 && y >= 0 }
```
Comment (jbanass) [inline 2024/04/script.rb:46]: We need to swap these two variables. The shape of objects within `coords` is not in `x, y` format, but `y, x` (as shown below)

Comment (TyJaYo) [reply 2024/04/script.rb:46]: This always trips me up. I want it to be cartesian but it's always going to be a bit flipped. At this point I'm just happy if I can get the row number to be the second, _y_, coordinate and the column number to be the first as _x_. So I guess in your case I would leave this and change the other bit(s).

File: 2024/04/input.txt
Comment (jbanass) [inline 2024/04/input.txt:1]: @jbanass add input files to .gitignore and purge others

File: 2024/04/script.rb (hunk 1 above)
Comment (jbanass) [reply 2024/04/script.rb:46]: I would gladly if all the other bits weren't replicated dozens of times as arguments 😂

**Pull request merged**